test/javascripts/acceptance/chat-keyboard-shortcuts-test.js

```diff
@@ -251,4 +251,26 @@ acceptance("Discourse Chat - Keyboard shortcuts", function (needs) {
       "modal dismissed after submitting link"
     );
   });
+
+  test("Dash key (-) opens chat float", async function (assert) {
+    await visit("/latest");
+    this.chatService.set("sidebarActive", false);
+    this.chatService.set("chatWindowFullPage", false);
+
+    await triggerKeyEvent(document.body, "keydown", "-");
+    await settled();
+    assert.ok(visible(".topic-chat-float-container"), "chat float is open");
+  });
+
+  test("Escape to close chat float", async function (assert) {
+    await visit("/latest");
+    this.chatService.set("sidebarActive", false);
+    this.chatService.set("chatWindowFullPage", false);
```

Comment:
Oh you missed the `triggerKeyEvent` for escape here, probably why the test failed

Reply:
Thanks @martin-brennan! 

I've added that back, but it's still failing the test unfortunately. I've tried using different selectors too. However, it doesn't seem to be working. Do you know what I might be missing?

Reply:
It might not be a test issue - does it work in the app? Global shortcuts don't work when input fields are focused. 😃

Reply:
Yeah it worked locally when I checked it out

Reply:
> Global shortcuts don't work when input fields are focused. 

They do, since it works for the chat composer's other shortcuts :) And it worked for the escape one locally too

Reply:
Ah, I got it backwards! This means `-` binding shouldn't be global then, otherwise typing this character in any input field will open the chat drawer, right?

Reply:
Oh yes you are right! That is a pickle :thinking: @keegangeorge you have to make sure there is no `document.activeElement` I guess when `-` is pressed so the chat drawer doesn't open if you are already focused on another input

Reply:
@CvX Thanks for catching this and @martin-brennan thanks for the advice on how to handle it. I used the `element.target` since the element was already being passed. Let me know what you think.